system/Config/Services.php

```diff
@@ -545,7 +544,7 @@ public static function routes(bool $getShared = true)
      *
      * @return Router
      */
-    public static function router(?RouteCollectionInterface $routes = null, ?Request $request = null, bool $getShared = true)
+    public static function router(?RouteCollection $routes = null, ?Request $request = null, bool $getShared = true)
```

Comment:
This makes it more difficult for people to pass their own classes that implement the `RouteCollectionInterface`, which they currently can, so this would be BC.

Reply:
The current Router does not depend on `RouteCollectionInterface`, it actually depends on `RouteCollection`.

If a user creates their own `RouteCollection` which implements `RouteCollectionInterface`,
it is not compatible with the `Router`, because some methods are missing in the `RouteCollectionInterface`.

Reply:
@lonnieezell can we pick this discussion back up?